lib/active_record/typed_store/behavior.rb

```diff
@@ -78,16 +85,16 @@ def attribute_names_for_partial_inserts
       # Contrary to all vanilla Rails types, typedstore attribute have an inherent default
       # value that doesn't match the database column default.
       # As such we need to insert them on partial inserts even if they weren't changed.
-      super | self.class.typed_stores.keys.map(&:to_s)
+      super | self.class.typed_store_attribute_names
     end
 
     def attribute_names_for_partial_updates
       # On partial updates we shouldn't need to force stores to be persisted. However since
       # we weren't persisting them for a while on insertion, we now need to gracefully deal
       # with existing records that may have been persisted with a `NULL` store
       # We use `blank?` as an heuristic to detect these.
-      super | self.class.typed_stores.keys.map(&:to_s).select do |store|
-        @attributes.key?(store) && @attributes[store].value_before_type_cast.blank?
+      super | self.class.typed_store_attribute_names.select do |store|
+        read_attribute_before_type_cast(store).blank?
```

Comment:
Actually, I was checking `@attribute.key?` in case it's a partially loaded model (I should have added a test for it TBH)